distributed/worker.py

```diff
@@ -1494,9 +1494,12 @@ async def start(self):
 
         setproctitle("dask-worker [%s]" % self.address)
 
-        await asyncio.gather(
+        plugins_msgs = await asyncio.gather(
             *(self.plugin_add(plugin=plugin) for plugin in self._pending_plugins)
         )
+        for msg in plugins_msgs:
+            if msg["status"] != "OK":
+                raise pickle.loads(msg["exception"].data)
```

Comment:
I'm wondering what happens if multiple plugins raise an exception. do we at least log all of them? If not, should we iterate over all, log them and only afterwards raise?

FWIW, I think this would be a great application for exception groups proposed (and accepted) in [PEP654](https://www.python.org/dev/peps/pep-0654/). It's not implemented, yet, just a heads up.

Reply:
Done in https://github.com/dask/distributed/pull/4298/commits/7552423149be6a123c35d6f642a969f976d0fbbc .